slog.go

```diff
@@ -0,0 +1,214 @@
+package btclog
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+)
+
+// Handler wraps the slog.Handler interface with a few more methods that we
+// need in order to satisfy the Logger interface.
+type Handler interface {
+	slog.Handler
+
+	// Level returns the current logging level of the Handler.
+	Level() Level
+
+	// SetLevel changes the logging level of the Handler to the passed
+	// level.
+	SetLevel(level Level)
+}
+
+// sLogger is an implementation of Logger backed by a structured sLogger.
+type sLogger struct {
+	Handler
+	logger *slog.Logger
+}
+
+// NewSLogger constructs a new structured logger from the given Handler.
+func NewSLogger(handler Handler) Logger {
+	return &sLogger{
+		Handler: handler,
+		logger:  slog.New(handler),
+	}
+}
+
+// Tracef formats message according to format specifier, prepends the prefix as
+// necessary, and writes to log with LevelTrace.
+//
+// This is part of the Logger interface implementation.
+func (l *sLogger) Tracef(format string, params ...any) {
+	l.logger.Log(context.Background(), slog.Level(LevelTrace),
```

Comment:
nit: could add an internal helper method that does the `context.Background()` and `fmt.Sprintf()` part?
Then just pass `l.toSLog(LevelTrace, format, params...)`.

Reply:
nice idea! added 💪